swan-lake/learn/running-ballerina-code.md

```diff
@@ -54,139 +47,92 @@ You can compile a source file with an entry point into an executable jar.
 
-### Ballerina Runtime Configuration Files
-
-A Ballerina runtime can be configured using configuration parameters, which are arbitrary key/value pairs with structure. The `ballerina/config` module provides an API for sourcing configuration parameters and using them within your source code. See [Config API Documentation](/swan-lake/learn/api-docs/ballerina/config/index.html) for details.
+### Ballerina Runtime Configurable Variables
 
-The configuration APIs accept a key and an optional default value. If a mapping does not exist for the specified key, the default value is returned as the configuration value. The default values of these optional configurations are the default values of the return types of the functions.
+A Ballerina runtime can be configured using configurable variables, which can be configured using `Config.toml` file. 
+`Config.toml` file may or may not provide configuration values for configurable variables that are used in the program. 
+Ballerina currently supports configurable variables of types `int`, `float`, `boolean`, `string`, `decimal` and the arrays of respective types. 
+See [Configurable BBE](/swan-lake/learn/by-example/configurable.html) for more details.
 
-### Sourcing Parameters into Ballerina Programs
-Configuration parameters for your programs and apps can be defined on the CLI, as an environment variable, or from a configuration file, with loading and override precedence in the same order.
-
-#### Sourcing CLI Parameters
-Consider the following example, which reads a Ballerina config value and prints it.
+Consider the following example, which uses configurable variables.
 
 ```ballerina
 import ballerina/io;
-import ballerina/config;
+
+configurable int id = ?;
+configurable string name = "Ann";
+configurable boolean married = true;
 
 public function main() {
-  string name = config:getAsString("hello.user.name");
-  io:println("Hello, " + name + " !");
+  io:println("User ID : ", id);
+  io:println("User Name : ", name);
+  io:println("Married : ", married);
 }
 ```
 
-The config key is `hello.user.name`. To pass a value to this config from the CLI, we can use `--key=value` format as the following command.
-```bash
-$ bal run  main.bal --hello.user.name=Ballerina
-Hello, Ballerina !
-```
-
-#### Sourcing Configuration Values
-
-The value can be passed as a config file as well. A configuration file should conform to the [TOML](https://github.com/toml-lang/toml) format. Ballerina only supports the following features of TOML: value types (string, int, float, and boolean), tables, and nested tables. Given below is a sample `ballerina.conf`:
-
+`?` denotes that `id` is a required configuration, hence `Config.toml` file must contain a value for key `id`.
+If a default value assigned, the configuration is optional, hence `Config.toml` file may or may not contain a values for
+configurable variables `name` and `married`.
+Consider the below `Config.toml` file.
 ```toml
-[hello.user]
-name="Ballerina"
+id = 1001
```

Comment:
We have a mandatory structure as 
```toml 
[module-name]
id = 111
``` 
Since we give a single bal file execution here, we might not need to add it. 
But don't we need to add this info somewhere? It's not in BBE. 
@warunalakshitha

Reply:
Could be covered in #1716

Reply:
IMO default case should be project.

Reply:
Even example we need use a project

Reply:
IMO since we need to keep learn pages as simple as possible it's better to use the basic here with a single `.bal` file and add the details in #1716

Reply:
@warunalakshitha @HindujaB WDYT?

Reply:
+1 Lets fix with config learn page.